libs/agent-runner.js

```diff
@@ -99,6 +101,91 @@ agent.prototype.runPushOperation = function runPushOperation(operation) {
     log.error('Warning: A primary key has not been set, which means rows will always be appended to the data source whenever this script runs. To allow updating rows, please define a primary key to be used for the comparison.');
   }
 
+  if (operation.encrypt) {
+    if (!_.get(operation, 'encrypt.fields', []).length) {
+      log.critical('[ENCRYPTION] You need to define a list of fields to encrypt.');
+    }
+
+    log.info(`[ENCRYPTION] Encryption is enabled for the following fields: ${operation.encrypt.fields.join(', ')}`);
+
+    if (operation.encrypt.key) {
+      log.debug('[ENCRYPTION] A key has been defined by the operation.');
+      encryptionKey = operation.encrypt.key;
+    } else {
+      log.debug('[ENCRYPTION] A key has not been defined. Fetching the key from Fliplet servers...');
+
+      const keySalt = operation.encrypt.salt || `dataSource-${operation.targetDataSourceId}`;
+
+      encryptionKey = await this.api.request({
+        url: `v1/data-sources?type=keystore`
+      }).then(async (response) => {
+        let dataSource;
+
+        if (!response.data.dataSources.length) {
+          log.debug('[ENCRYPTION] Generating new keystore...');
+
+          const organizations = await this.api.request({
+            url: 'v1/organizations'
+          }).then((response) => response.data.organizations);
+
+          const organization = _.first(organizations);
+
+          log.debug(`Fetched organization ${organization.name}`);
+
+          const result = await this.api.request({
+            method: 'POST',
+            url: 'v1/data-sources',
+            data: {
+              name: 'Keystore',
+              type: 'keystore',
```

Comment:
is this a new type?

Reply:
Yes.

Reply:
what does it mean?

Reply:
It means we can apply specific rules given this type and it's also not shown in the data source manager.